Flaky test failure on remote fetcher tests

I'm having a problem or would like to suggest a feature.

My current problem is that we have a flaky test failure under jruby during remote fetcher tests.

The error is like the following (I temporaritly enabled verbose output to help debugging):

```
(... more output ...)
TestGemRemoteFetcher#test_ssl_connection = 0.33 s = .
TestGemRemoteFetcher#test_fetch_s3_instance_profile_creds_with_region = 0.05 s = .
TestGemRemoteFetcher#test_ssl_client_cert_auth_connection = 0.04 s = S
TestGemRemoteFetcher#test_cache_update_path = 0.04 s = .
TestGemRemoteFetcher#test_fetch_path_openssl_ssl_sslerror = 0.04 s = .
TestGemRemoteFetcher#test_download_local = 0.03 s = .
TestGemRemoteFetcher#test_fetch_s3_no_host = 0.03 s = .
TestGemRemoteFetcher#test_observe_no_proxy_env_single_host = 0.08 s = .
TestGemRemoteFetcher#test_download_local_read_only = 0.05 s = .
TestGemRemoteFetcher#test_download_unsupported = 0.04 s = .
TestGemRemoteFetcher#test_download_read_only = 0.04 s = .
TestGemRemoteFetcher#test_download_same_file = 0.05 s = .
TestGemRemoteFetcher#test_fetch_s3_instance_profile_creds_with_token = 0.04 s = .
TestGemRemoteFetcher#test_cache_update_path_no_update = 0.03 s = .
/home/runner/.rvm/rubies/jruby-9.2.9.0/lib/ruby/stdlib/webrick/https.rb:50: warning: javax.net.ssl.SSLPeerUnverifiedException: peer not authenticated
Bad file descriptor - No message available
rake aborted!
Command failed with status (1)

Tasks: TOP => test
(See full trace by running task with --trace)
TestGemRemoteFetcher#test_do_not_follow_insecure_redirect = 
##[error]Process completed with exit code 1.
```

Sample failing workflows:
* https://github.com/rubygems/rubygems/pull/3103/checks?check_run_id=409632584.
* https://github.com/rubygems/rubygems/runs/411671791
* https://github.com/rubygems/rubygems/runs/418567152
* https://github.com/rubygems/rubygems/runs/488030252

This issue is related to:

  - [ ] Network problems
  - [ ] Installing a library
  - [ ] Publishing a library
  - [ ] The command line `gem`
  - [x] Other

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

--- a/lib/rubygems/test_utilities.rb
+++ b/lib/rubygems/test_utilities.rb
@@ -40,7 +40,7 @@ class Gem::FakeFetcher
   end
 
   def find_data(path, nargs = 3)
-    return File.read path.path if URI === path and 'file' == path.scheme
+    return Gem.read_binary path.path if URI === path and 'file' == path.scheme
 
     if URI === path and "URI::#{path.scheme.upcase}" != path.class.name
       raise ArgumentError,

